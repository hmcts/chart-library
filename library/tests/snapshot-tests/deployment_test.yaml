suite: Snapshot tests
templates:
  - templates/v2/deployment.yaml
tests:
  - it: manifest should match snapshot
    asserts:
      - matchSnapshot: {}