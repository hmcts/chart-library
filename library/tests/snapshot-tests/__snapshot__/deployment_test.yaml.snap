manifest should match snapshot:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      annotations: null
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: release-name-library
        helm.sh/chart: library-0.1.0
      name: release-name-library
    spec:
      replicas: 2
      revisionHistoryLimit: 0
      selector:
        matchLabels:
          app.kubernetes.io/name: release-name-library
      template:
        metadata:
          annotations: null
          labels:
            app.kubernetes.io/instance: RELEASE-NAME
            app.kubernetes.io/managed-by: Helm
            app.kubernetes.io/name: release-name-library
            helm.sh/chart: library-0.1.0
        spec:
          affinity:
            podAntiAffinity:
              requiredDuringSchedulingIgnoredDuringExecution:
                - labelSelector:
                    matchExpressions:
                      - key: app.kubernetes.io/name
                        operator: In
                        values:
                          - release-name-library
                  topologyKey: kubernetes.io/hostname
          containers:
            - args:
                - this
                - that
              command:
                - this
                - that
              env:
                - name: STORAGE_ACCOUNT_NAME
                  valueFrom:
                    secretKeyRef:
                      key: storageAccountName
                      name: storage-secret-example-release-name
                - name: REFORM_ENVIRONMENT
                  value: preview
                - name: REFORM_SERVICE_NAME
                  value: rhubarb-frontend
                - name: REFORM_TEAM
                  value: cnp
              envFrom:
                - configMapRef:
                    name: release-name-library
                - secretRef:
                    name: env-secret
              image: hmctspublic.azurecr.io/chart-nodejs/testapp
              imagePullPolicy: Always
              livenessProbe:
                failureThreshold: 3
                httpGet:
                  path: /health/liveness
                  port: 3000
                initialDelaySeconds: 5
                periodSeconds: 15
                timeoutSeconds: 3
              name: release-name-library
              ports:
                - containerPort: 3000
                  name: http
              readinessProbe:
                httpGet:
                  path: /health
                  port: 3000
                initialDelaySeconds: 5
                periodSeconds: 15
                timeoutSeconds: 3
              resources:
                limits:
                  cpu: 500m
                  memory: 512Mi
                requests:
                  cpu: 25m
                  memory: 64Mi
              securityContext:
                allowPrivilegeEscalation: false
              startupProbe:
                failureThreshold: 3
                httpGet:
                  path: /health/liveness
                  port: 3000
                initialDelaySeconds: 5
                periodSeconds: 15
                timeoutSeconds: 3
              volumeMounts:
                - mountPath: /mnt/secrets/bulk-scan
                  name: vault-bulk-scan
                  readOnly: true
                - mountPath: /run/secrets/bulk-scan-alias
                  name: vault-bulk-scan-v2
                  readOnly: true
          dnsConfig:
            options:
              - name: ndots
                value: "3"
          restartPolicy: Always
          securityContext:
            fsGroup: 1000
            runAsUser: 1000
          serviceAccountName: NAMESPACE
          terminationGracePeriodSeconds: 30
          volumes:
            - csi:
                driver: secrets-store.csi.k8s.io
                readOnly: true
                volumeAttributes:
                  secretProviderClass: release-name-library-bulk-scan
              name: vault-bulk-scan
            - csi:
                driver: secrets-store.csi.k8s.io
                readOnly: true
                volumeAttributes:
                  secretProviderClass: release-name-library-bulk-scan-v2
              name: vault-bulk-scan-v2
